provisioning/miniprov/hare_mp/main.py

```diff
@@ -565,6 +565,14 @@ def add_param_argument(parser):
     return parser
 
 
+def add_factory_argument(parser):
+    parser.add_argument('--pre-factory',
```

Comment:
from the document https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/221642873/CORTX+Factory+Manufacturing+Process, I see some confusion,
```
Perform Factory Reset
Testing may have generated a log Login using CORTX cluster CLI. All the data and configuration needs to be cleared. After you are done with all the tests you can perform the factory reset using the following commands. 


$ cortx_cli cluster reset [--type {data|all}] [--pre-factory]
Note: 

**This internally calls <component>_setup reset --config <URL> and <component>_setup cleanup--config <URL> for all components.**

Normally factory reset will bring the node up to the factory configured stage. There will be an additional option provided to undo “post_install” activities as well. For example,  Motr LVM configuration will be removed when “--factory” is passed. By default deployment logs are not cleaned up but they will be cleaned up when --factory is passed to it. 

Node will marked with appropriate stage.
```
But Hare HLD for the same mentions,
```
Note here that '--pre-factory' option here will be an additional option provided to undo “post_install” activities. For more info refer to above doc

Hare’s factory reset(reset + cleanup) currently do not handle this option.

We need to support this option in future(
```
So, few questions,
1. will `--pre-factory` option be passed to the component's setup utility? or is it for provisioner and provisioner will internally invoke reset + cleanup?
2. is the option `--pre-factory` or --factory` as mentioned in the factory manufacturing process document?
3. If hare needs to support `--pre-factory` then, do we need to invoke `reset + cleanup` as part of `--pre-factory` implementation? cc @SwapnilGaonkar7 , @vaibhavparatwar .

Reply:
`<component>_setup cleanup --config <URL> --pre-factory` this command needs to be supported for pre-factory.
pre-factory is an additional option to cleanup interface to undo post-install phase.
We don't need to call to reset and cleanup as pre-factory is an additional option for the cleanup phase.
@mssawant Please let me know if it answers your questions. Ref. https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/559743237/F-97+Reset+Cleanup+Pre-Factory

Reply:
Yes it does. Thank you.